workflows/rnaseq/main.nf

```diff
@@ -785,6 +785,105 @@ workflow RNASEQ {
     emit:
     multiqc_report = ch_multiqc_report // channel: /path/to/multiqc_report.html
     versions       = ch_versions       // channel: [ path(versions.yml) ]
+
+    publish:
+    QUANTIFY_STAR_SALMON.out                            >> 'star_salmon'
+    QUANTIFY_STAR_SALMON.out.multiqc                    >> null
+    QUANTIFY_STAR_SALMON.out.merged_counts_transcript   >> null
+    QUANTIFY_STAR_SALMON.out.merged_tpm_transcript      >> null
+    QUANTIFY_STAR_SALMON.out.versions                   >> null
+
+    // TODO: !params.skip_alignment && params.aligner == 'star_salmon' && params.with_umi
+    // BAM_SORT_STATS_SAMTOOLS.out.bam         >> (params.save_align_intermeds || params.save_umi_intermeds ? 'star_salmon' : null)
+    // BAM_SORT_STATS_SAMTOOLS.out.bai         >> (params.save_align_intermeds || params.save_umi_intermeds ? 'star_salmon' : null)
+    // BAM_SORT_STATS_SAMTOOLS.out.stats       >> (params.save_align_intermeds || params.save_umi_intermeds ? 'star_salmon/samtools_stats' : null)
+    // BAM_SORT_STATS_SAMTOOLS.out.flagstat    >> (params.save_align_intermeds || params.save_umi_intermeds ? 'star_salmon/samtools_stats' : null)
+    // BAM_SORT_STATS_SAMTOOLS.out.idxstats    >> (params.save_align_intermeds || params.save_umi_intermeds ? 'star_salmon/samtools_stats' : null)
+    // SAMTOOLS_SORT.out.bam                   >> (params.save_align_intermeds || params.save_umi_intermeds ? 'star_salmon' : null)
+    // UMITOOLS_PREPAREFORSALMON.out.bam       >> (params.save_align_intermeds || params.save_umi_intermeds ? 'star_salmon' : null)
+    // UMITOOLS_PREPAREFORSALMON.out.log       >> (params.save_align_intermeds || params.save_umi_intermeds ? 'star_salmon/umitools/log' : null)
+
+    // TODO: !params.skip_alignment && params.aligner == 'star_salmon'
+    DESEQ2_QC_STAR_SALMON.out               >> 'star_salmon/deseq2_qc'
+    DESEQ2_QC_STAR_SALMON.out.pca_multiqc   >> null
+    DESEQ2_QC_STAR_SALMON.out.dists_multiqc >> null
+    DESEQ2_QC_STAR_SALMON.out.versions      >> null
+
+    // TODO: !params.skip_alignment && params.aligner == 'star_rsem'
+    // DESEQ2_QC_RSEM.out               >> 'star_rsem/deseq2_qc'
+    // DESEQ2_QC_RSEM.out.pca_multiqc   >> null
+    // DESEQ2_QC_RSEM.out.dists_multiqc >> null
+    // DESEQ2_QC_RSEM.out.versions      >> null
```

Comment:
Already discussed this before, but channels like these need to be prepared in the main body if the process is conditionally executed, so that the process `.out` is only accessed in the conditional block, for example:

```groovy
main:
def deseq2_qc_rsem = Channel.empty()
if( !params.skip_alignment && params.aligner == 'star_rsem' ) {
  DESEQ2_QC_RSEM( /* ... */ )
  deseq2_qc_rsem = Channel.empty().mix(
    DESEQ2_QC_RSEM.out.pdf,
    DESEQ2_QC_RSEM.out.rdata,
    // ...
  )
}

publish:
deseq2_qc_rsem >> 'star_rsem/deseq2_qc'